### PR TITLE
Feature/user/login

### PR DESCRIPTION
**Pull request by DobocanDarius**

### CODE REVIEW

File: InternshipProject#2/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿using InternshipProject_2.Helpers;
+using InternshipProject_2.Manager;
+using InternshipProject_2.Models;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
```
Comment (iiusan) [inline InternshipProject#2/Controllers/UserController.cs:9]: this is not SOLID!!!
why do you have references to entity framework?

File: InternshipProject#2/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace InternshipProject_2.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        private readonly IUserManager _userManager;
+        private readonly Token _token;
+        private readonly PasswordHash _hash;
```
Comment (iiusan) [inline InternshipProject#2/Controllers/UserController.cs:22]: where is this used?

File: InternshipProject#2/Helpers/Token.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+using InternshipProject_2.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace InternshipProject_2.Helpers;
+
+public class Token
```
Comment (iiusan) [inline InternshipProject#2/Helpers/Token.cs:11]: looks like this class it's a model. needs a rename

File: InternshipProject#2/Manager/UserManager.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+using Microsoft.EntityFrameworkCore;
+using RequestResponseModels.User.Request;
+using RequestResponseModels.User.Response;
+
+namespace InternshipProject_2.Manager;
+
+public class UserManager : IUserManager
+{
+    private readonly Project2Context _context;
+    private readonly PasswordHash _hash;
```
Comment (iiusan) [inline InternshipProject#2/Manager/UserManager.cs:15]: where is this used?

File: RequestResponseModels/User/Request/LoginRequest.cs [hunk 5]
```diff
@@ -8,15 +8,17 @@ namespace RequestResponseModels.User.Request;
...
-        Username = username;
-     Password = password;
         Email = email;
+        Password = password;
+
     }
-    public string Username { get; set; } = null!;
+
 
     public string Password { get; set; } = null!;
```
Comment (iiusan) [inline RequestResponseModels/User/Request/LoginRequest.cs:19]: without null!

File: RequestResponseModels/User/Request/UserRequest.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RequestResponseModels.User.Request
+{
+    public class UserRequest
```
Comment (iiusan) [inline RequestResponseModels/User/Request/UserRequest.cs:9]: empty class

File: RequestResponseModels/User/Response/UserResponse.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RequestResponseModels.User.Response;
+
+public class UserResponse
```
Comment (iiusan) [inline RequestResponseModels/User/Response/UserResponse.cs:9]: empty class

**Review by iiusan — CHANGES_REQUESTED** [review]

File: RequestResponseModels/User/Response/LoginResponse.cs [hunk 8]
```diff
@@ -1,23 +1,16 @@
...
-
-    public string Username { get; set; } = null!;
-
-    public string Password { get; set; } = null!;
-
-    public string Email { get; set; } = null!;
-
-    public DateTime CreatedAt { get; set; }
+    public string? Token { get; set; }
+    public IEnumerable<LoginRequest> Users { get; set; }
```
Comment (hunorszekely) [inline RequestResponseModels/User/Response/LoginResponse.cs:9]: Not sure why are you returning this... Why is it a collection?

File: InternshipProject#2/Manager/UserManager.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public async Task<LoginResponse> Login(LoginRequest user)
+    {
+        var foundUser = await _context.Users.FirstOrDefaultAsync(u => u.Email == user.Email && u.Password == _hash.HashPassword(user.Password));
+
+        if (foundUser != null)
+        {
+            var map = MapperConfig.InitializeAutomapper();
+            var loginResponse = map.Map<LoginResponse>(foundUser);
```
Comment (hunorszekely) [inline InternshipProject#2/Manager/UserManager.cs:30]: Have you tested this mapping? It cannot map the user into the LoginResponse.

File: InternshipProject#2/Manager/UserManager.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            loginResponse.Token = _token.Generate(foundUser);
```
Comment (hunorszekely) [inline InternshipProject#2/Manager/UserManager.cs:31]: If you are using AutoMapper, all the properties should be mapped inside of it.

File: InternshipProject#2/Program.cs [hunk 11]
```diff
@@ -1,21 +1,45 @@
...
+            ValidateIssuerSigningKey = true,
+            IssuerSigningKey = new SymmetricSecurityKey(key)
+        };
+    });
 builder.Services.AddScoped<ICommentManager, CommentManager>();
 builder.Services.AddScoped<ITicketManager, TicketManager>();
 builder.Services.AddDbContext<Project2Context>();
 builder.Services.AddScoped<IAssigneeManager, AssigneeManager>();
 
+
```
Comment (hunorszekely) [inline InternshipProject#2/Program.cs:42]: Pay attention to empty lines as well.

**Review by hunorszekely — CHANGES_REQUESTED** [review]

**Review by hunorszekely — APPROVED** [review]

**Pull request merged**